rust-runtime/aws-smithy-runtime/src/client/http/body/minimum_throughput/throughput.rs

```diff
@@ -0,0 +1,262 @@
+        if self.per_time_elapsed == Duration::from_secs(1) {
+            // Already normalized to per second, return self without modifying it.
+            self
+        } else {
+            // Normalize to per second and return the result.
+            Self {
+                bytes_read: self.bytes_per_second(),
+                per_time_elapsed: Duration::from_secs(1),
+            }
+        }
+    }
+}
+
+impl PartialEq for Throughput {
+    fn eq(&self, other: &Self) -> bool {
+        if self.per_time_elapsed == other.per_time_elapsed {
+            // If both are using the same unit of time, compare bytes per second.
+            self.bytes_per_second() == other.bytes_per_second()
+        } else {
+            // Otherwise, normalize the unit of time and then compare adjusted bytes per second.
+            let normalized_self = self.normalize_to_per_one_second();
+            let normalized_other = other.normalize_to_per_one_second();
+            normalized_self.bytes_per_second() == normalized_other.bytes_per_second()
+        }
+    }
+}
+
+impl PartialOrd for Throughput {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        if self.per_time_elapsed == other.per_time_elapsed {
+            // If both are using the same unit of time, compare bytes per second.
+            self.bytes_per_second()
+                .partial_cmp(&other.bytes_per_second())
+        } else {
+            // Otherwise, normalize the unit of time and then compare adjusted bytes per second.
+            let normalized_self = self.normalize_to_per_one_second();
+            let normalized_other = other.normalize_to_per_one_second();
+            normalized_self
+                .bytes_per_second()
+                .partial_cmp(&normalized_other.bytes_per_second())
+        }
+    }
+}
+
+impl fmt::Display for Throughput {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // The default float formatting behavior will ensure the a number like 2.000 is rendered as 2
+        // while a number like 0.9982107441748642 will be rendered as 0.9982107441748642. This
+        // multiplication and division will truncate a float to have a precision of no greater than 3.
+        // For example, 0.9982107441748642 would become 0.999. This will fail for very large floats
+        // but should suffice for the numbers we're dealing with.
+        let pretty_bytes_per_second = (self.bytes_per_second() * 1000.0).round() / 1000.0;
+
+        write!(f, "{pretty_bytes_per_second} B/s")
+    }
+}
+
+impl From<(u64, Duration)> for Throughput {
+    fn from(value: (u64, Duration)) -> Self {
+        Self {
+            bytes_read: value.0 as f64,
+            per_time_elapsed: value.1,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(super) struct ThroughputLogs {
+    max_length: usize,
+    min_elapsed_time: Duration,
+    inner: VecDeque<(SystemTime, u64)>,
+    floating_back: Option<(SystemTime, u64)>,
```

Comment:
what's floating_back?

Reply:
ignore it, I'm removing it in favor of a simpler strategy.